### PR TITLE
Add booking bug api

### PR DESCRIPTION
**Pull request by dwhenry**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,12 +4,21 @@ ruby '2.3.0'
...
 group :development do
   gem 'rake'
 end
 
+group :development, :test do
+  gem 'pry-byebug'
+end
+
 group :test do
+  gem 'rspec'
```
Comment (andrewgarner) [inline Gemfile]: Can we keep the dependencies in alphabetical order? It keeps things easier to read when our dependencies grow!

File: lib/booking_bug/api.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        url = data['_links']['next'] && data['_links']['next']['href']
+      end
+
+      bookings
+    end
+
+    private
+
+    def base_path
+      "/api/v1/admin/#{ENV['BOOKING_BUG_COMPANY']}/bookings"
```
Comment (andrewgarner) [inline lib/booking_bug/api.rb]: `BookingBug::Api` doesn't need to decide how it should be configured (in this case using `ENV`). By doing so we've coupled the class to `ENV`.

If we were to introduce a "company" attribute we can remove the coupling. We can set a sensible default to make using the class easier too.

```rb
class BookingBug::API
  def initialize(company_id: ENV['BOOKING_BUG_COMPANY'])
    self.company_id = company_id
  end

  attr_accessor :company_id

  private

  def base_path
    "/api/v1/admin/#{company_id}/bookings"
  end
end
```

File: lib/booking_bug/api.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class BookingBug
+  class UnableToAuthenticate < StandardError; end
+
+  class Api
```
Comment (andrewgarner) [inline lib/booking_bug/api.rb]: Maybe this should be `AdminAPI`?

I'm not sure there is any value in abstracting the fact that this is class is an adapter for the admin API, not all Booking Bug APIs.

File: lib/booking_bug/connection.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          req.headers['App-Id'] = app_id
+          req.headers['App-Key'] = api_key
+          req.headers['Auth-Token'] = auth_token
+        end
+      )
+    end
+
+    private
+
+    def parse(response)
```
Comment (andrewgarner) [inline lib/booking_bug/connection.rb]: There is a Faraday middleware that will automatically parse JSON response types.

File: lib/booking_bug/api.rb (hunk 3 above)
Comment (dwhenry) [inline lib/booking_bug/api.rb]: I was planning to leave this until we get to the second implementation and look to refactor/scope these things as made sense then.

At this stage I am unsure of what otehr parts will exist or how they will stick together.

**Comment by dwhenry** [comment]
Call BookingBug.extract to get array of booking bug data. Example record:
```
{
  booked_on: Date.new(2015, 03, 11),
  metadata: { booking_id: 3 }
}
```

**Pull request closed without merging**